ffmpeg/api_test.go

```diff
@@ -877,3 +877,123 @@ func consecutiveMuxerOpts(t *testing.T, accel Acceleration) {
 	run(cmd)
 
 }
+
+func TestTranscoder_EncodingProfiles(t *testing.T) {
+	encodingProfiles(t, Software)
+}
+
+func encodingProfiles(t *testing.T, accel Acceleration) {
+	run, dir := setupTest(t)
+	defer os.RemoveAll(dir)
+	cmd := `
+        cp "$1/../transcoder/test.ts" test.ts
+    `
+	run(cmd)
+
+	// First, sanity check defaults
+	for _, p := range VideoProfileLookup {
+		if p.Profile != ProfileNone {
+			t.Error("Default VideoProfile profile not set to ProfileNone")
+		}
+	}
+
+	// Encode to all H264 profiles
+	var profilesMap = map[Profile]string{
```

Comment:
It might be good to write this test in a way that it fails out if we ever add more profiles to the `ProfileParameters ` mapping. Something like this:

```golang
var profilesMap map[Profile]string
for v, _ := ProfileParameters {
  switch v {
    case ProfileNone:
      // skip
    case ProfileH264Baseline:
      profilesMap[v] = "baseline"
    ... check rest of profiles ...
    default: 
      t.Error("Unhandled profile ", v)
  }
}
```